internal/datacoord/task_scheduler.go

```diff
@@ -47,9 +49,10 @@ type taskScheduler struct {
 	collectMetricsDuration time.Duration
 
 	// TODO @xiaocai2333: use priority queue
-	tasks      map[int64]Task
-	notifyChan chan struct{}
-	taskLock   *lock.KeyLock[int64]
+	tasks        map[int64]Task
+	notifyChan   chan struct{}
+	taskLock     *lock.KeyLock[int64]
+	failTaskChan chan UniqueID
```

Comment:
remove this? 
Just provide a CancelTaskByID and use key lock to modify corresponding task state.